File: core/src/ru/myitschool/dcrawler/entities/skills/Battlecry.java
package ru.myitschool.dcrawler.entities.skills;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.utils.Array;
import ru.myitschool.dcrawler.effects.BattlecryEffect;
import ru.myitschool.dcrawler.entities.Entity;
import ru.myitschool.dcrawler.math.DiceAction;
import ru.myitschool.dcrawler.math.MathAction;
import ru.myitschool.dcrawler.entities.skills.patterns.CharacterTargetPattern;
import ru.myitschool.dcrawler.entities.skills.patterns.TargetPattern;
import ru.myitschool.dcrawler.entities.skills.targeting.TargetDisplayer;

/**
 * Created by Voyager on 18.11.2017.
 */
public class Battlecry extends Skill {

    int accuracyBonus = 5;
    MathAction action = new DiceAction(6);

    public Battlecry(Entity doer) {
        super(doer);
    }

    @Override
    public void maintainDisplayers(Array<TargetDisplayer> displayers) {

    }

    @Override
    public TargetPattern getPattern() {
        return new CharacterTargetPattern(this);
    }

    @Override
    public Texture getIcon() {
        return new Texture("battle_cry.png");
    }

    @Override
    public String getName() {
        return "Battle cry";
    }

    @Override
    public String getDescription() {
        return "You gets additional +" + accuracyBonus + " accuracy and additional "
                + action.getDescription() + " damage for all attack on your next turn";
    }

    @Override
    public int getSkillAccuracyBonus() {
        return 0;
    }

    @Override
    public int getRange() {
        return 0;
    }

    @Override
    public int getDistanceMin() {
        return 0;
    }

    @Override
    public int getDistanceMax() {
        return 0;
    }

    @Override
    public int getTargetCountMax() {
        return 0;
    }

    @Override
    public int getCooldownMax() {
        return 0;
    }

    @Override
    public int getType() {
        return SKILL_TYPE_AT_WILL;
    }

    @Override
    public int getTargetType() {
        return SKILL_TARGET_TYPE_SELF;
    }

    @Override
    public boolean isCheckAllTargets() {
        return false;
    }

    @Override
    public boolean isMarkEverything() {
        return false;
    }

    @Override
    public boolean isMark() {
        return false;
    }

    @Override
    public boolean isObstruct() {
        return false;
    }

    @Override
    public boolean isWallTargets() {
        return false;
    }

    @Override
    public void use() {
        getDoer().addEffect(new BattlecryEffect(getDoer(), accuracyBonus, action, 1));
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/Battlecry.kt
package com.wolg_vlad.dcrawler.entities.skills

import com.badlogic.gdx.graphics.Texture
import com.wolg_vlad.dcrawler.effects.BattlecryEffect
import com.wolg_vlad.dcrawler.entities.Entity
import com.wolg_vlad.dcrawler.entities.skills.patterns.CharacterTargetPattern
import com.wolg_vlad.dcrawler.entities.skills.patterns.TargetPattern
import com.wolg_vlad.dcrawler.entities.skills.targeting.TargetDisplayer
import com.wolg_vlad.dcrawler.math.DiceAction
import com.wolg_vlad.dcrawler.math.MathAction

/**
 * Created by Voyager on 18.11.2017.
 */
class Battlecry(doer: Entity) : Skill(doer) {
    var accuracyBonus = 5
    var action: MathAction = DiceAction(6)

    override val displayers: Array<TargetDisplayer> = arrayOf()
    override val pattern: TargetPattern = CharacterTargetPattern(this)
    override val icon: Texture = Texture("battle_cry.png")
    override val name: String = "Battle cry"
    override val description: String = "You gets additional +$accuracyBonus accuracy and additional " +
            "${action.description} damage for all attack on your next turn"
    override val skillAccuracyBonus: Int = 0
    override val range: Int = 0
    override val distanceMin: Int = 0
    override val distanceMax: Int = 0
    override val targetCountMax: Int = 0
    override val cooldownMax: Int = 0
    override val type: Int = SKILL_TYPE_AT_WILL
    override val targetType: Int = SKILL_TARGET_TYPE_SELF
    override val isCheckAllTargets: Boolean = false
    override val isMarkEverything: Boolean = false
    override val isMark: Boolean = false
    override val isObstruct: Boolean = false
    override val isWallTargets: Boolean = false

    override fun use() {
        doer.addEffect(BattlecryEffect(doer, accuracyBonus, action, 1))
    }
}